nav2_util/include/nav2_util/service_client.hpp

```diff
@@ -37,22 +37,36 @@ class ServiceClient
     } else {
       node_ = generate_internal_node(service_name + "_Node");
```

Comment:
I know this is not part of your PR, but if this really allows us to fix what you propose, you can remove the default node parameter requiring `generate_internal_node`, and require that a user provide a node to use. I looked through the codebase and there's 1 test you'd need to update/remove but otherwise that should work.

Reply:
I just saw that the `ClearEntirelyCostmapServiceClient` inherit from `ServiceClient` without provided node.
But this Class is never used. Should I remove it?